Library/Homebrew/Gemfile

```diff
@@ -70,6 +70,9 @@ end
 group :vscode, optional: true do
   gem "ruby-lsp", require: false
 end
+group :debugger, optional: true do
+  gem "debug"
+end
```

Comment:
This already ships with Portable Ruby btw:

```
$ brew ruby -rdebug/version -e 'puts DEBUGGER__::VERSION'
1.9.1
```

Reply:
Ohhhh, that explains why I kept getting weird version mismatch issues where 1.9.1 was used somewhere and 1.9.2 other places. Will it be a problem to rely on the shipped version because it won't be updated, or not really

Reply:
> Will it be a problem to rely on the shipped version because it won't be updated, or not really

It'll get updated roughly every Ruby 3.y: https://stdgems.org/debug/. We don't really need it to be any more frequent.

We had a similar discussion when using it for `brew tests --debug`. We decided to use the bundled version as pulling it in as a dependency requires libyaml to be installed on the system (which is the CI failure you're seeing) which gets very messy to try enforce that through all of our CI.

The catch is because it's a bundled gem rather than a default gem - it usually isn't available under a Bundler environment, so we rely on something we insert into Portable Ruby: `/opt/homebrew/Library/Homebrew/vendor/portable-ruby/current/lib/ruby/3.3.0/arm64-darwin20/portable_ruby_gems.rb` (loaded from https://github.com/Homebrew/brew/blob/master/Library/Homebrew/standalone/init.rb#L51).

So it doesn't work with system Ruby on Linux, but this is OK because debugging is only for us anyway and not for end users. Just make the command fail if `HOMEBREW_USING_PORTABLE_RUBY` is false.

> I can't remember, everyone is forced to use portable ruby now, right?

On macOS yes. On Linux it will use a system Ruby if it's new enough but we perhaps should use Portable Ruby more aggressively.

On aarch64 Linux, Portable Ruby isn't available.

Reply:
> Just make the command fail if `HOMEBREW_USING_PORTABLE_RUBY` is false.

Unresolving until this is done!